azurelinuxagent/common/protocol/wire.py

```diff
@@ -528,17 +524,20 @@ class WireClient(object):
 
     def __init__(self, endpoint):
         logger.info("Wire server endpoint:{0}", endpoint)
-        self.endpoint = endpoint
-        self.goal_state = None
-        self.hosting_env = None
-        self.shared_conf = None
-        self.remote_access = None
-        self.certs = None
-        self.ext_conf = None
-        self.host_plugin = None
+        self._endpoint = endpoint
```

Comment:
Made almost all WireClient properties private as they shouldn't be directly accessed. They should only be accessed by getter/setters